### ISSUE TITLE
Does TopologyProposal use positions?

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#102]
Does `TopologyProposal` actually make use of `old_positions` or `new_positions`?  I don't think we actually use these anywhere since we split out the geometry engine.

Can we eliminate these?

**Comment by jchodera** [comment]
We would still need positions passed to `GeometryEngine.propose()`, but it could be a separate argument:
```python
new_positions, geometry_logp  = self.geometry_engine.propose(topology_proposal, old_positions)
```

**Comment by pgrinaway** [comment]
Again, this was for the possibility of position-dependent proposals.

**Comment by jchodera** [comment]
We can leave it for now, but it leads to really weird constructs that require us to replace the `positions` in `TopologyProposal`:
https://github.com/choderalab/perses/blob/master/perses/tests/test_sampler.py#L151-L166

**Comment by pgrinaway** [comment]
Oh that is weird, you're right. I hadn't thought about that.

**Comment by jchodera** [comment]
We could change the NCMC API, but I'm not sure which is better.

**Comment by pgrinaway** [comment]
Better to change the `ProposalEngine` I think.

**Comment by pgrinaway** [comment]
Addressed in #106
